sum_types_generator/lib/src/templates.dart

```diff
@@ -141,6 +141,36 @@ String function({
         ";",
     ].join("\n");
 
+String expressionFunction({
+  bool isStatic = false,
+  required String type,
+  required String name,
+  Iterable<TypeParamSpec> typeParams = const [],
+  Iterable<String> posParams = const [],
+  Iterable<String> namedParams = const [],
+  Iterable<String>? body,
+}) =>
+    [
+      if (isStatic) "static ",
+      type,
+      " ",
+      name,
+      parametrize(typeParams),
+      "(",
+      ...posParams.map(appendComma),
+      if (namedParams.isNotEmpty) ...[
+        "{",
+        ...namedParams.map(appendComma),
+        "}",
+      ],
+      ")",
+      if (body != null) ...[
```

Comment:
An expression-function cannot be empty, so the condition can be removed and the type of `body` changed to non-nullable. The standard function can be something like `void f() {}`, but and expression must always return something (even if it's `void`, e.g. `void f() => print("x")`).